vpr/src/place/manual_move_generator.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * @file 	manual_move_generator.cpp
+ * @author	Paula Perdomo
+ * @date 	2021-07-19
+ * @brief 	Contains the ManualMoveGenerator class memeber definitions.
+ */
+
+#include "manual_move_generator.h"
+#include "manual_moves.h"
+
+ManualMoveGenerator::ManualMoveGenerator(std::unique_ptr<SoftmaxAgent>& agent) {
```

Comment:
Looks like we make a duplicate set of move generators (all of them) just to override the propose_move function for the manual move generator.  @MohamedElgammal : this seems rather heavyweight to make a new propose_move function; is this the right design pattern?

@PaulaPerdomo : comment what the functions in this file are doing, and why.